game-core/src/system/camera.rs

```diff
@@ -0,0 +1,48 @@
+use amethyst::renderer::Camera; 
+use amethyst::{
+    core::Transform,
+    ecs::{Join, ReadStorage, System, WriteStorage},
+};
+
+use crate::component::Player;
+
+pub struct Movement;
+
+impl<'s> System<'s> for Movement {
+    type SystemData = (
+        ReadStorage<'s, Player>,
+        ReadStorage<'s, Camera>,
+        WriteStorage<'s, Transform>,
+    );
+
+    fn run(&mut self, (players, cameras, mut transforms): Self::SystemData) {
+        let mut player_x: f32 = 0.0;
+        let mut player_y: f32 = 0.0;
+
+        // let mut camera_x: f32 = 0.0;
+        // let mut camera_y: f32 = 0.0;
+
+        // let mut camera_safe_x: f32 = 0.0;
+        // let mut camera_safe_y: f32 = 0.0;
+
+        for (_, transform) in (&players, &mut transforms).join() {
+            player_x = transform.translation.x - 256.0 as f32;// * 5.0;
+            player_y = transform.translation.y - 256.0 as f32;// * 5.0;
+        }
+
+        for (_, transform) in (&cameras, &mut transforms).join() {
+            let camera_x = transform.translation.x as f32;// * 5.0;
```

Comment:
do these need to be converted? My understanding is that these types are already f32, I think these `as f32` can mostly be removed. That's the rust equivalent of a static cast between variable types, so you can write `let x: f32 = (0u32 as f32);` and it will compile, note if you change either (but not both) f32 to f64 it wont compile.